src/datadetection.cs

```diff
@@ -0,0 +1,123 @@
+using CoreFoundation;
+using ObjCRuntime;
+using Foundation;
+
+using System;
+
+namespace DataDetection {
+
+	[iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface DDDetectedValue
+	{
+		[Export ("matchedString")]
+		string MatchedString { get; }
+
+		[Export ("matchedRange")]
+		NSRange MatchedRange { get; }
+	}
+
+	[iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (DDDetectedValue))]
```

Comment:
base does not have a default ctor
I don't see how having one here would be helpful (since you can't set any of the values)